File: app/src/main/java/com/example/exchangingprivatelessons/domain/model/LessonRequest.kt
package com.example.exchangingprivatelessons.domain.model

data class LessonRequest(
    val id: String = "",
    val lessonId: String,
    val ownerId: String,        // lesson owner
    val requesterId: String,    // student asking for the lesson

    val status: RequestStatus,
    val requestedAt: Long,
    val respondedAt: Long? = null,
)

enum class RequestStatus { Pending, Approved, Declined }

File: functions/src/models/LessonRequest.ts
import type { FirestoreDataConverter } from "firebase-admin/firestore";
import { TS } from "./common";

export enum RequestStatus {
  Pending  = "Pending",
  Approved = "Approved",
  Declined = "Declined",
}

export interface LessonRequest {
  lessonId: string;
  ownerId: string;
  requesterId: string;
  status: RequestStatus;
  requestedAt: TS;
  respondedAt: TS | null;
}

export const lessonRequestConverter: FirestoreDataConverter<LessonRequest> = {
  toFirestore: (r) => r,
  fromFirestore: (snap) => snap.data() as LessonRequest,
};

